service/src/main.rs

```diff
@@ -491,6 +499,25 @@ fn start_worker<E, T, D>(
 		})
 		.unwrap();
 
+	// ------------------------------------------------------------------------
+	// Start polling loop to wait until we have a worker for a shard registered on
+	// the parentchain (TEEREX WorkerForShard). This is the pre-requisite to be
+	// considered initialized and ready for the next worker to start.
+	let node_api_for_initialized = node_api.clone();
+	let shard_for_initialized = *shard;
+	thread::spawn(move || {
+		loop {
+			if let Ok(Some(_)) =
+				node_api_for_initialized.worker_for_shard(&shard_for_initialized, None)
+			{
+				// Set that the service is initialized.
+				initialization_handler.worker_for_shard_registered();
+				break
+			}
+			thread::sleep(Duration::from_secs(2));
+		}
+	});
```

Comment:
I guess it would be useful to log once in the loop that we are still polling; otherwise, the worker stays just silent.

Reply:
Yes, I can do that 👍 will add a `println!` as well, once we successfully end the polling.

Reply:
Cool thanks!